exec/java-exec/src/main/java/org/apache/drill/exec/planner/cost/DrillRelMdDistinctRowCount.java

```diff
@@ -180,7 +180,7 @@ private Double getDistinctRowCountInternal(TableScan scan, RelMetadataQuery mq,
       }
     }
     s = Math.min(s, rowCount);
-    if (!allCols) {
+    if (s == 1.0 && !allColsHaveNDV) {
```

Comment:
Its not clear why the check for `s == 1.0` is needed.  
Also, I see that the `else` part is returning the selectivity value whereas the `if` part is returning a row count.  The method is expected to return a rowcount, not selectivity...unless I am missing something.

Reply:
`s` is rowcount not selectivity.

Reply:
Ah,  yes..I somehow mistook that for selectivity.  What about the check for `s == 1.0` ? Even if that value is higher but at least 1 column does not have NDV stats, I would think we want to return the 10% default.

Reply:
Renamed the variable to avoid confusion